module/data/shared/uses-field.mjs

```diff
@@ -157,62 +157,111 @@ export default class UsesField extends SchemaField {
   /**
    * Rolls a recharge test for an Item or Activity that uses the d6 recharge mechanic.
    * @this {Item5e|Activity}
-   * @returns {Promise<Roll|void>}
+   * @param {BasicRollProcessConfiguration} config   Configuration information for the roll.
+   * @param {BasicRollDialogConfiguration} dialog    Configuration for the roll dialog.
+   * @param {BasicRollMessageConfiguration} message  Configuration for the roll message.
+   * @returns {Promise<BasicRoll[]|void>}            The created Roll instances, or `null` if no die was rolled.
    */
-  static async rollRecharge() {
+  static async rollRecharge(config={}, dialog={}, message={}) {
     const uses = this.system ? this.system.uses : this.uses;
     const recharge = uses?.recovery.find(({ period }) => period === "recharge");
     if ( !recharge ) return;
 
-    const rollConfig = {
-      formula: "1d6",
-      data: this.getRollData(),
-      target: parseInt(recharge.formula),
-      chatMessage: true
-    };
+    const rollConfig = foundry.utils.mergeObject({
+      rolls: [{
+        parts: ["1d6"],
+        data: this.getRollData(),
+        options: {
+          target: parseInt(recharge.formula)
+        }
+      }]
+    }, config);
+    rollConfig.subject = this;
+
+    const dialogConfig = foundry.utils.mergeObject({ configure: false }, dialog);
+
+    const messageConfig = foundry.utils.mergeObject(({
+      create: true,
+      data: {
+        speaker: ChatMessage.getSpeaker({ actor: this.actor, token: this.actor.token })
+      },
+      preCreate: (rolls, rollConfig, messageConfig) => {
```

Comment:
I'm having some trouble following this design. This logic is specific to rolling recharges, so it should exist in the `rollRecharge` method. I don't see the benefit of adding an extra layer of indirection by passing a callback to `BasicRoll` here.

```js
if ( messageConfig.create ) {
  messageConfig.create = false;
  const [roll] = await CONFIG.Dice.BasicRoll.build(rollConfig, dialogConfig, messageConfig);
  if ( roll.isSuccess ) {
    // Perform updates ...
    roll.toMessage({ flavor: /* ... */, speaker: /* ... */ });
  }
}
```